### PR TITLE
test case: Adding gingko tests for block posture

### LINKED ISSUE #1062: Adding test cases for setting namespace based security posture
## Feature Request

**Short Description**

Current setting of namespace based security posture is not tested in our testing framework. This will help in improving and testing this feature in our CI

**Is your feature request related to a problem? Please describe the use case.**

Related to the issue #801, for [namespace based security posture](https://github.com/kubearmor/KubeArmor/wiki/KubeArmor-version-release-v0.3#default-security-posture-daemon1024)

**Describe the solution you'd like**

Expand Ginkgo framework for automated tests for setting security postures on a namespace level. This might require some extension in the util library for ginkgo.

**Describe alternatives you've considered**

Just checking for posture annotations using ` kubectl describe ns namespace` but it would be better if Kubearmor is able to tell the default security posture during the tests.

### PR DESCRIPTION
**Pull request by PrimalPimmy**
Signed-off-by: PrimalPimmy <Prashant20.pm@gmail.com>

**Purpose of PR?**:

Fixes #1062 
This Ginkgo test will test for all test cases when the Default Security Posture is set to block. I am using namespace annotations to set posture to block in the wordpress-mysql namespace.

**Does this PR introduce a breaking change?**
No.

**If the changes in this PR are manually verified, list down the scenarios covered:**:
This PR currently only has network based block posture set. Will add a File posture test as well.

**Checklist:**
- [x] Commit has unit tests
- [x] Commit has integration tests

### CODE REVIEW

File: tests/block/block_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			// Apply policy
+			err := K8sApplyFile("res/ksp-wordpress-allow-tcp.yaml")
+			Expect(err).To(BeNil())
+
+			// Start Kubearmor Logs
+			err = KarmorLogStart("policy", "wordpress-mysql", "Network", wp)
+			Expect(err).To(BeNil())
+
+			// wait for policy creation
+			time.Sleep(5 * time.Second)
```
Comment (rksharma95) [inline tests/block/block_test.go]: sleep not required here as well as after deleting policies in `AfterEach` section. the sleep is already added to the `K8sApplyFile()`.

Comment (PrimalPimmy) [reply tests/block/block_test.go]: okay :+1:

File: tests/block/block_test.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		// sql = getWpsqlPod("mysql-",
+		// 	"container.apparmor.security.beta.kubernetes.io/mysql: localhost/kubearmor-wordpress-mysql-mysql-mysql")
+	})
+
+	AfterEach(func() {
+		KarmorLogStop()
+		err := DeleteAllKsp()
+		Expect(err).To(BeNil())
+		// wait for policy deletion
+		time.Sleep(5 * time.Second)
```
Comment (rksharma95) [inline tests/block/block_test.go:68]: it can be removed. not needed

File: tests/block/block_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		It("can whitelist certain files accessed by a package while blocking all other sensitive content", func() {
+			// Apply policy
+			err := K8sApplyFile("res/ksp-wordpress-allow-file.yaml")
+			Expect(err).To(BeNil())
+
+			// Start Kubearmor Logs
+			err = KarmorLogStart("policy", "wordpress-mysql", "File", wp)
+			Expect(err).To(BeNil())
+
+			//curl needs UDP for DNS resolution
```
Comment (rksharma95) [inline tests/block/block_test.go:107]: irrelevant comment can we remove or update it.

File: tests/block/block_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+			sout, _, err := K8sExecInPod(wp, "wordpress-mysql", []string{"bash", "-c", "cat wp-config.php"})
+			Expect(err).To(BeNil())
+			fmt.Printf("---START---\n%s---END---\n", sout)
+			Expect(sout).To(MatchRegexp("cat.*Permission denied"))
+
+			//test that tcp is whitelisted
```
Comment (rksharma95) [inline tests/block/block_test.go:113]: can we remove/update this too

**Comment by Prateeknandle** [comment]
@PrimalPimmy are these tests running locally? Failing for me actually. Not getting alerts generated.
I'm having k3s cluster, kubearmor, wordpress-mysql app, anything else we need here?

Update - both tests passed

**Comment by Prateeknandle** [comment]
@PrimalPimmy current tests seems ok, can we also add one test for capabilities.

**Comment by PrimalPimmy** [comment]
@Prateeknandle do we have capabilities support yet?

**Comment by Prateeknandle** [comment]
> @Prateeknandle do we have capabilities support yet?

oh sorry didn't know about that, then the pr is fine

File: tests/block/block_test.go (hunk 2 above)
Comment (PrimalPimmy) [reply tests/block/block_test.go:68]: you mean to delete all 4 lines or the sleep? @rksharma95

Comment (rksharma95) [reply tests/block/block_test.go:68]: sleep only

**Review by Ankurk99 — COMMENTED** [review]
PR looks fine to me, but have we tested this?

**Comment by PrimalPimmy** [comment]
@Ankurk99 me and @Prateeknandle tested it locally and it works.

File: tests/block/block_test.go (hunk 2 above)
Comment (PrimalPimmy) [reply tests/block/block_test.go:68]: @rksharma95 the CI fails if I remove this sleep function

Comment (PrimalPimmy) [reply tests/block/block_test.go:68]: but it works locally.. I don't get it

Comment (PrimalPimmy) [reply tests/block/block_test.go:68]: https://github.com/kubearmor/KubeArmor/actions/runs/4187120477/jobs/7256865710 this is failing, Im reverting back to adding sleep

Comment (rksharma95) [reply tests/block/block_test.go:68]: it was just a flake as you can see it's passing for docker and crio. anyways we can keep it :+1:

**Review by nyrahul — APPROVED** [review]

**Pull request merged**